great_expectations/data_context/data_context/abstract_data_context.py

```diff
@@ -58,25 +136,560 @@ def __init__(self, runtime_environment: dict):
+
+        Why does this exist in AbstractDataContext? CloudDataContext and FileDataContext both use it
+
+        """
+        if hasattr(self, "_context_root_directory"):
+            return self._context_root_directory
+        return
+
+    @property
+    def project_config_with_variables_substituted(self) -> DataContextConfig:
+        return self.get_config_with_variables_substituted()
+
+    @property
+    def plugins_directory(self) -> Optional[str]:
+        """The directory in which custom plugin modules should be placed.
+
+        Why does this exist in AbstractDataContext? CloudDataContext and FileDataContext both use it
+        """
+        return self._normalize_absolute_or_relative_path(
+            self.project_config_with_variables_substituted.plugins_directory
+        )
+
+    @property
+    def stores(self) -> dict:
+        """A single holder for all Stores in this context"""
+        return self._stores
+
+    @property
+    def expectations_store_name(self) -> Optional[str]:
+        return self.project_config_with_variables_substituted.expectations_store_name
+
+    @property
+    def expectations_store(self) -> ExpectationsStore:
+        return self.stores[self.expectations_store_name]
+
+    @property
+    def evaluation_parameter_store_name(self):
+        return (
+            self.project_config_with_variables_substituted.evaluation_parameter_store_name
+        )
+
+    @property
+    def evaluation_parameter_store(self):
+        return self.stores[self.evaluation_parameter_store_name]
+
+    @property
+    def validations_store_name(self):
+        return self.project_config_with_variables_substituted.validations_store_name
+
+    @property
+    def validations_store(self) -> ValidationsStore:
+        return self.stores[self.validations_store_name]
+
+    @property
+    def checkpoint_store_name(self):
+        try:
+            return self.project_config_with_variables_substituted.checkpoint_store_name
+        except AttributeError:
+            from great_expectations.data_context.store.checkpoint_store import (
+                CheckpointStore,
+            )
+
+            if CheckpointStore.default_checkpoints_exist(
+                directory_path=self.root_directory
+            ):
+                return DataContextConfigDefaults.DEFAULT_CHECKPOINT_STORE_NAME.value
+            if self.root_directory:
+                error_message: str = f'Attempted to access the "checkpoint_store_name" field with no `checkpoints` directory.\n  Please create the following directory: {os.path.join(self.root_directory, DataContextConfigDefaults.DEFAULT_CHECKPOINT_STORE_BASE_DIRECTORY_RELATIVE_NAME.value)}\n  To use the new "Checkpoint Store" feature, please update your configuration to the new version number {float(CURRENT_GE_CONFIG_VERSION)}.\n  Visit https://docs.greatexpectations.io/docs/guides/miscellaneous/migration_guide#migrating-to-the-batch-request-v3-api to learn more about the upgrade process.'
```

Comment:
I'd prefer to see thing string broken up and meet our line length limit unless we have a policy about keeping strings greppable (or some other policy). Python will concatenate multiple line strings for you. You can prepend all lines or the specific line you want with an `f` if you want to interpolate:
```
    error_message: str = ('Attempted to access the "checkpoint_store_name" field '
                                          f'with no `checkpoints` directory.\n ...'
```

I also don't think the `str` annotation is necessary since construction is happening in this line and the correct type will be inferred. I know we like to be explicit so 🤷 .

Reply:
This one was a bit of a journey, but I was able to format this message along with other ones in `abstract_data_context` that were super long.